### PR TITLE
Spline survey integration

### PR DESCRIPTION
**Pull request by squilter**
It works.  Things that are missing:

- [x] A different icon in the fragment_editor_list
- [x] A fragment_editor_detail.  Should I try to reuse the one for Survey?

### CODE REVIEW

**Comment by ne0fhyk** [comment]
Yea reuse the king from the survey for the fragment editor detail.
For the icon, pick the most fitting from https://materialdesignicons.com

--
Fredia Huya-Kouadio
Android Tech Lead, 3D Robotics
fredia@3dr.com

**Comment by squilter** [comment]
I drew a nice new icon.  How do I import it so that it scales it and puts it in all the right folders?

**Comment by ne0fhyk** [comment]
@squilter i usually rely on gimp for icon rescaling.

**Comment by squilter** [comment]
I remember an importer tool.  Maybe it was eclipse.  It would take a high quality image and scale it multiple times and put it in all the right folders.  Do you know what that is?

**Comment by squilter** [comment]
@ne0fhyk can you upload 2.4.7 so that Travis passes?  Then this is ready for merge.

Making the SplineSurvey render as spline waypoints would require some changes to MissionProxy getPathPoints().  I couldn't figure it out.

**Comment by squilter** [comment]
fixes #1459

File: Android/src/org/droidplanner/android/proxy/mission/item/fragments/MissionSurveyFragment.java [hunk 1]
```diff
@@ -178,10 +183,11 @@ public void onScrollingEnded(CardWheelHorizontalView cardWheel, Object startValu
                             surveyDetail.setSidelap(mSidelapPicker.getCurrentValue());
                         }
 
-                        final MissionItem.ComplexItem<Survey>[] surveys = surveyList
+                        final MissionItem.ComplexItem<T>[] surveys = surveyList
                                 .toArray(new MissionItem.ComplexItem[surveyList.size()]);
 
                         drone.buildMissionItemsAsync(surveys, this);
+                        System.out.println("nothing");
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/proxy/mission/item/fragments/MissionSurveyFragment.java]: @squilter forgot that behind. should be removed.

Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/proxy/mission/item/fragments/MissionSurveyFragment.java]: @squilter nevermind, i'll take care of it.

**Comment by ne0fhyk** [comment]
@squilter Looks good for merging. I'll wait until the required version for the `dronekit-android` client library is made public, then merge into `develop`.
Good job!

**Comment by ne0fhyk** [comment]
Closing! I'll be reopened this pr against the `release-3.1.7` branch.

**Pull request closed without merging**